File: app/src/main/java/br/com/hussan/githubapi/GithubApp.java
package br.com.hussan.githubapi;

import android.app.Application;

/**
 * Created by hussan on 10/10/16.
 */

public class GithubApp extends Application {

    AppComponent component;

    @Override
    public void onCreate()
    {
        super.onCreate();

        component = DaggerAppComponent.builder()
                    .appModule(new AppModule(this))
                    .build();
    }

    public AppComponent getComponent()
    {
        return component;
    }
}

File: app/src/main/kotlin/br/com/hussan/githubapi/GithubApp.kt
package br.com.hussan.githubapi

import android.app.Application
import br.com.hussan.githubapi.data.api.ApiModule

/**
 * Created by hussan on 10/10/16.
 */

class GithubApp : Application() {

    lateinit var component: AppComponent

    override fun onCreate() {
        super.onCreate()

        component = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .apiModule(ApiModule())
                .build()
    }
}
